File: shared-types/src/models/Task.ts
import { UserResponse } from './User';

export interface ITask {
  _id: string;
  title: string;
  description?: string;
  boardId: string;
  columnId: string;
  position: number;
  assignee?: string | UserResponse;
  priority?: 'low' | 'medium' | 'high';
  dueDate?: Date;
  tags?: string[];
  createdBy: string | UserResponse;
  createdAt: Date;
  updatedAt: Date;
}

export interface TaskResponse {
  _id: string;
  title: string;
  description?: string;
  boardId: string;
  columnId: string;
  position: number;
  assignee?: UserResponse;
  priority?: 'low' | 'medium' | 'high';
  dueDate?: Date;
  tags?: string[];
  createdBy: UserResponse;
  createdAt: Date;
  updatedAt: Date;
}

export interface TaskListResponse {
  success: boolean;
  data: TaskResponse[];
  count: number;
}

File: backend-fastapi/app/models/task.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class UserResponse(BaseModel):
    id: str = Field(alias='_id')
    email: str
    username: str

    class Config:
        populate_by_name = True


class TaskResponse(BaseModel):
    id: str = Field(alias='_id')
    title: str
    description: Optional[str] = None
    boardId: str
    columnId: str
    position: int
    assignee: Optional[UserResponse] = None
    priority: Optional[str] = None
    dueDate: Optional[datetime] = None
    tags: Optional[list[str]] = None
    createdBy: UserResponse
    createdAt: datetime
    updatedAt: datetime

    class Config:
        populate_by_name = True


class TaskListResponse(BaseModel):
    success: bool
    data: list[TaskResponse]
    count: int
